src/pymmcore_widgets/mda/_core_mda.py

```diff
@@ -161,8 +170,28 @@ def _on_run_clicked(self) -> None:
         ):
             return
 
+        sequence = self.value()
+
+        # if the z plan is relative, and there are multiple stage positions, but the
+        # 'include z' is unchecked, raise an error since we cannot run a
+        # relative z plan without knowing the starting z for each position.
+        if (
+            sequence.z_plan
+            and sequence.z_plan.is_relative
+            and len(sequence.stage_positions) > 1
+            and not self.stage_positions.include_z.isChecked()
+        ):
+            QMessageBox.critical(
+                self,
+                "Z Plan Error",
+                "Cannot run a relative Z plan without including a starting Z position. "
```

Comment:
generally speaking, in useq-schema, the absence of a value (such as a Z position) is to be interpreted as "use the current value".  Why not use the current z position here?

Reply:
True, I've updated the value method with this logic.